### PR TITLE
Gitlab reporter

### PR DESCRIPTION
**Pull request by cmouse**
- Does not actually seem to do anything during start of build, not sure why.

### CODE REVIEW

File: master/docs/relnotes/index.rst [hunk 1]
```diff
@@ -44,7 +46,7 @@ Deprecations, Removals, and Non-Compatible Changes
 * The buildmaster now requires at least Twisted-14.0.1.
 
 * GitLabStatus was present in buildbot 0.8.12 release, but is no more in 0.9.0 (:bug:`3286`).
```
Comment (tardyp) [inline master/docs/relnotes/index.rst]: actually it wasn't as we saw yesterday on IRC. You can drop this remark.
You can say:

* new :bb:reporter:`GitLabStatusPush` to report builds results to gitlab.

**Comment by tardyp** [comment]
👍 great work!

File: master/buildbot/reporters/gitlab.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from buildbot.process.properties import Interpolate
```
Comment (sa2ajj) [inline master/buildbot/reporters/gitlab.py]: could you please re-order imports:

1. Python runtime
2. 3rd party modules
3. Buildbot modules

**Comment by sa2ajj** [comment]
Yay, thank you very much!

**Comment by tardyp** [comment]
arg. I did not realize there were no unit tests :-/

**Comment by cmouse** [comment]
Ow.  I can see if I could produce some.

**Comment by tardyp** [comment]
@cmouse 
I created some tests here: https://github.com/buildbot/buildbot/pull/2468

I'd appreciate if you can test the resulting code.

**Pull request merged**